docs/docs/reference/02-yaml-syntax.md

```diff
@@ -0,0 +1,638 @@
+
+Components can be loaded or unloaded by component name.
+
+```yaml
+load: <component_name>
+unload: <component_name>
+```
+
+It is possible to load or unload multiple components simultaneously by specifying a list of names.
+
+```yaml
+load:
+  - component_a
+  - component_b 
+```
+
+#### Transition from one component to another
+
+Component A can invoke a transition to component B as a shorthand for "unload component A, load component B".
+
+```yaml
+transition: <component_name>
+```
+
+#### Trigger a lifecycle transition
+
+
+```yaml
+# 
+  lifecycle: "configure"
+```
+
+Request a lifecycle transition on the component that is triggering the event, using one of the available transitions
+(`configure`, `activate`, `deactivate`, `cleanup`, or `shutdown`).
+```yaml
+lifecycle: activate
+```
+
+Request a lifecycle transition on a different component.
+```yaml
+lifecycle:
+  transition: activate
+  component: <component_name>
+```
+
+Use a list to trigger multiple transitions from a single predicate.
+```yaml
+lifecycle:
+  - transition: activate
+    component: <component_name>
+  - transition: deactivate
+    component: <component_name>
+```
+
+#### Set a parameter
+
+Set a parameter on the component that is triggering the event.
+```yaml
+set:
+  parameter: <parameter_name>
+  value: <parameter_value>
+```
+
+Set a parameter on a different component.
+```yaml
+set:
+  parameter: <parameter_name>
+  value: <parameter_value>
+  component: <component_name>
+```
+
+Set a parameter on the controller of a particular hardware interface.
+```yaml
+set:
+  parameter: <parameter_name>
+  value: <parameter_value>
+  controller: <controller_name>
+  interface: <hardware_interface_name>
+```
+
+#### Call a service
+
+Call a service with no payload on the component that is triggering the event.
+```yaml
+service: <service_name>
+```
+
+Call a service on a different component.
+```yaml
+service:
+  name: <service_name>
+  component: <component_name>
+```
+
+Call a service with a string payload.
+```yaml
+service:
+  name: <service_name>
+  component: <component_name>
+  payload: "..."
+```
+
+The service payload can also be written as any standard YAML object. The application parser will automatically encode
+the object into a string format when making the service call. In this case, the component service is responsible
+for parsing the string back into a YAML object, dict or structure as necessary.
+
+```yaml
+service:
+  name: <service_name>
+  component: <component_name>
+  payload:
+    foo: "some content"
+    bar: [ x, y, z ]
+    baz:
+      a: 1
+      b: 2
+```
+
+#### Load or unload a hardware interface
+
+Load and initialize a hardware interface.
+```yaml
+load_hardware: <hardware_interface_name>
+```
+
+Unload and destroy a hardware interface.
+```yaml
+unload_hardware: <hardware_interface_name>
+```
+
+:::caution
+
+All hardware interfaces in the application are automatically loaded and initialized when the application starts.
```

Comment:
Wait is that true or do you want it to be like that?

Reply:
Well yes that is essentially true at the moment (auto-load hardware has always been default true in all of our use-cases). The service and hardware refactor will give us a bit more fine-grained control over initial and subsequent hardware states in future, but for now this is a fair summary.